### PR TITLE
FSSS-125 Feat: Creates new tokens and Styling Documentation

### PR DESCRIPTION
**Pull request by renatamottam**
## What's the purpose of this pull request?
Create and restructure some tokens in order to improve the customization. Also, this PR adds a new documentation on our README related to the section "Styling Components".

### 🖊️ Styling Components

Our customized theme is based on [Design Tokens](https://css-tricks.com/what-are-design-tokens/) using [CSS Variables](https://developer.mozilla.org/en-US/docs/Web/CSS/--*) of a `class` for each one of them. Today, we have the following files in the `src/styles` folder:

### `global.scss`

Here you'll find the basic structure to build your theme (font base, color palette, spacing, color-text, body background color...), feel free to update it with your brand guidelines.

#### <b>Colors</b>

We suggest using a color palette of 3 colors and its gradation: `primary`, `secondary` and `neutral`.

We also listed a couple of customizable tokens so you can easily change your body background, for example.

If you feel the need to edit some of the color decisions, you can enter `colors.scss` and update the semantical tokens. E.g.:

```scss
--color-border-input: var(--color-neutral-4); // Current
--color-border-input: var(--color-neutral-5); // Updated
```

#### <b>Typography</b>

We use the [Modular Scale](https://www.modularscale.com/) setting to create our text-sizes. If you want to change it, just set the `base-font-size` and the `scale` ratio.

#### <b>Spacing</b>

The spacing scale is based on `rem` sizes, so it will remain consistent if you change the `base-font-size`.

### `grid.scss`

List of classes used to create default page grid.

```scss
.grid-content-full // Should be used for sections that are side to side, generally with a colored background.
.grid-content // Should be used for sections that fit centered on the grid.
.grid-section // This class only adds default vertical margins for page sections.
```

image

### `typography.scss`

For the typography-related tokens, we decided to use classes to add extra stylings like `font-weight` and `line-height`. In this file, you'll see all the classes for titles, paragraphs, and default settings on the body. You can create new ones here if needed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,50 +31,50 @@ As of Dec, 22, 2021, this starter is still far from covering most basic cases fo
...
+   Your site is now running at `http://localhost:8000`!
 
-> Note: You'll also see a second link: _`http://localhost:8000/___graphql`_. This is a tool you can use to experiment with querying your data. Learn more about using this tool in the [Gatsby tutorial](https://www.gatsbyjs.com/docs/tutorial/part-4/#use-graphiql-to-explore-the-data-layer-and-write-graphql-queries). Also, if you want to query for dynamic data (allProducts, allCollections, search etc), you can use tools like [GraphQL Playground](https://github.com/graphql/graphql-playground) and configure it to query _`http://localhost:8000/api/graphql`_
+> Note: You'll also see a second link: \_`http://localhost:8000/___graphql`_. This is a tool you can use to experiment with querying your data. Learn more about using this tool in the [Gatsby tutorial](https://www.gatsbyjs.com/docs/tutorial/part-4/#use-graphiql-to-explore-the-data-layer-and-write-graphql-queries). Also, if you want to query for dynamic data (allProducts, allCollections, search etc), you can use tools like [GraphQL Playground](https://github.com/graphql/graphql-playground) and configure it to query _`http://localhost:8000/api/graphql`\_
```
Comment (filipewl) [inline README.md:77]: Why did you add the `\`? This bit looks weirder now.

Before|After
-|-
CleanShot 2022-01-17 at 12 34 37|CleanShot 2022-01-17 at 12 33 32

File: src/styles/global.scss [hunk 2]
```diff
@@ -77,7 +82,7 @@
   --page-padding-notebook: var(--space-5);
 
   // Target Size
-  --min-size: var(--space-7);
+  --min-size: 48px;
```
Comment (filipewl) [inline src/styles/global.scss:85]: Maybe not for this PR, but throwing it in here since the thought just came to me. 

I think we should rename this variable to be more specific, to something like **tap-min-size**. If I hadn't seen the comment above, I wouldn't know it's about the minimum size required for _tappable_/clickable elements on mobile.

File: src/styles/typography.scss [hunk 3]
```diff
@@ -10,11 +10,16 @@
   --text-size-5: calc(var(--text-size-4) * var(--scale));    // 31px
   --text-size-6: calc(var(--text-size-5) * var(--scale));    // 39px
   --text-size-7: calc(var(--text-size-6) * var(--scale));    // 48px
-  --text-size-8: calc(var(--text-size-7) * var(--scale));    // 61px
+  --text-size-8: calc(var(--text-size-7) * var(--scale));    // 56px
```
Comment (filipewl) [inline src/styles/typography.scss:13]: This should be 61px, not 56px.
😮

CleanShot 2022-01-17 at 12 54 43

File: stylelint.config.js [hunk 4]
```diff
@@ -32,6 +32,7 @@ module.exports = {
     'selector-pseudo-class-no-unknown': true,
     'selector-pseudo-element-no-unknown': true,
     'comment-no-empty': true,
+    'max-line-length': 200,
```
Comment (filipewl) [inline stylelint.config.js:35]: 200 is a big number for the maximum line length. Is there a particular reason 200 was chosen? Common values for coding are 80 (maybe too little) or 120. Maybe we can go with 120 if 80 isn't good enough? I couldn't find out what was the default one used by Stylelint if no none was specified.

File: README.md [hunk 5]
```diff
@@ -302,18 +310,51 @@ function MySection {
 ```
 
 ## 🖊️ Styling Components
-This starter uses [Tailwind CSS](https://tailwindcss.com/) for styling. If you want, you can remove it and use other solutions. Both Gatsby and Store UI support many different CSS frameworks, like [emotion](https://emotion.sh/docs/introduction), [stitches](https://stitches.dev/docs/introduction) or even no CSS framework at all. Check the supported frameworks at [the Gatsby website](https://www.gatsbyjs.com/docs/how-to/styling/built-in-css/#other-css-options)
-This guide covers best practices and patterns to use when styling with Tailwind.
 
-Tailwind is a utility-first CSS framework. The goal of this starter is to make it possible for developers to use Tailwind themes to change the look of the store. To accomplish this, a few things need to be respected:
+Our customized theme is based on [Design Tokens](https://css-tricks.com/what-are-design-tokens/) using [CSS Variables](https://developer.mozilla.org/en-US/docs/Web/CSS/--*) of a `class` for each one of them. Today, we have the following files in the `src/styles` folder:
```
Comment (filipewl) [inline README.md:314]: I couldn't understand the first sentence very well, particularly the part "of a `class` for each one of them". When you say "class", is that a CSS class selector, or maybe JS's `class` construct?
:\

File: README.md (hunk 1 above)
Comment (renatamottam) [reply README.md:77]: This was an automatic fix by our Lint. Should I move it back?

File: src/styles/global.scss (hunk 2 above)
Comment (renatamottam) [reply src/styles/global.scss:85]: Agreed!

File: README.md (hunk 5 above)
Comment (renatamottam) [reply README.md:314]:
```suggestion
Our customized theme is based on [Design Tokens](https://css-tricks.com/what-are-design-tokens/) using [CSS Variables](https://developer.mozilla.org/en-US/docs/Web/CSS/--*) or a `css class` for each one of them. Today, we have the following files in the `src/styles` folder:
```

File: stylelint.config.js (hunk 4 above)
Comment (renatamottam) [reply stylelint.config.js:35]: I need this in order to add this mixin: [input-focus-ring](https://github.com/vtex-sites/base.store/pull/224/files#diff-866f5ecb98e17251bb80a1e6b2acf44f290b595a7ff29dbf29f15d6be250192aR25
)

File: src/styles/global.scss (hunk 2 above)
Comment (hellofanny) [reply src/styles/global.scss:85]: Agreed!

File: src/styles/typography.scss (hunk 3 above)
Comment (renatamottam) [reply src/styles/typography.scss:13]: True!

File: src/styles/global.scss (hunk 2 above)
Comment (renatamottam) [reply src/styles/global.scss]: ✅

File: stylelint.config.js (hunk 4 above)
Comment (filipewl) [reply stylelint.config.js:35]: Ah, I see now.

I see 2 options then:
1. Use line breaks for that line, e.g.
```css
box-shadow: 0 0 0 1px var(--bg-neutral-lightest),
    0 0 0 var(--border-width-2) var(--bg-focus-ring),
    inset 0 0 0 var(--border-width-0) var(--color-border-input-selected);
```

2. Temporarily disable Stylelint for that specific line, e.g.
```scss
@mixin focus-ring {
  outline: none;
  /* stylelint-disable-next-line max-line-length */
  box-shadow: 0 0 0 1px var(--bg-neutral-lightest), 0 0 0 var(--border-width-2) var(--bg-focus-ring);
}
```

I like the #1 better. What do you think? Then we don't need to modify this config.

File: README.md (hunk 5 above)
Comment (filipewl) [reply README.md:314]: I like the following formatting for "CSS class"  better:
```suggestion
Our customized theme is based on [Design Tokens](https://css-tricks.com/what-are-design-tokens/) using [CSS Variables](https://developer.mozilla.org/en-US/docs/Web/CSS/--*) or a CSS class for each token. Today, we have the following files in the `src/styles` folder:
```

File: README.md (hunk 1 above)
Comment (filipewl) [reply README.md:77]: That's weird! I tried changing this file but I didn't have the same result, the linter didn't touch this line. There must be some misconfiguration. 
😕 

I suggest reverting this change specifically.

File: README.md [hunk 6]
```diff
@@ -212,19 +215,21 @@ import type { ButtonProps } from '@faststore/ui'
...
   return <UIButton {...props} />
 }
 
 export default Button
 ```
 
 Now, your Button component is powered by Store UI. However, if you try to use this on your app you will see that the button is lacking styles. To add styles, we will use CSS modules because they allow us to target data attributes. On your terminal, type:
+
 ```sh
 touch src/components/ui/Button/Button.module.css
```
Comment (hellofanny) [inline README.md:228]:
```suggestion
touch src/components/ui/Button/Button.scss
```

Comment (hellofanny) [reply README.md:228]: If they keep our currently pattern, it would be a `.scss` file.

File: README.md [hunk 7]
```diff
@@ -212,19 +215,21 @@ import type { ButtonProps } from '@faststore/ui'
...
 ```
 
 Now, on `Button.module.css`:
+
 ```css
 [data-store-button] {
   @apply p-0 bg-primary-100;
```
Comment (hellofanny) [inline README.md:235]: We need to remove it and add another example.

File: README.md [hunk 8]
```diff
@@ -302,18 +310,51 @@ function MySection {
...
+
+The spacing scale is based on `rem` sizes, so it will remain consistent if you change the `base-font-size`.
+
+### `grid.scss`
+
+List of classes used to create default page grid.
+
+```scss
+.grid-content-full // Should be used for sections that are side to side, generally with a colored background.
+.grid-content // Should be used for sections that fit centered on the grid.
```
Comment (hellofanny) [inline README.md:347]: Not sure if this it's clear enough. 
An image here would be helpful 🤔

**Review by hellofanny — CHANGES_REQUESTED** [review]
Thanks for reviewing this section! 🤩🤩🤩
LGTM! Just left some comments to remove/update old code.

File: README.md [hunk 9]
```diff
@@ -212,19 +215,21 @@ import type { ButtonProps } from '@faststore/ui'
...
 export default Button
 ```
 
 Now, your Button component is powered by Store UI. However, if you try to use this on your app you will see that the button is lacking styles. To add styles, we will use CSS modules because they allow us to target data attributes. On your terminal, type:
+
 ```sh
 touch src/components/ui/Button/Button.module.css
 ```
 
 Now, on `Button.module.css`:
```
Comment (hellofanny) [inline README.md:231]:
```suggestion
Now, on `Button.scss`:
```

File: src/styles/global.scss [hunk 10]
```diff
@@ -77,7 +82,7 @@
   --page-padding-notebook: var(--space-5);
 
   // Target Size
-  --min-size: var(--space-7);
+  --tap-min-size: 48px;
```
Comment (hellofanny) [inline src/styles/global.scss:85]: Niceeee!

File: README.md (hunk 8 above)
Comment (filipewl) [reply README.md:347]: Agreed!

**Comment by filipewl** [comment]
Allow me to merge `main-styled` to fix the existing conflict and also attempt to fix the "Something Went Wrong" for the Build.

CleanShot 2022-01-18 at 10 05 03

File: README.md (hunk 8 above)
Comment (eduardoformiga) [reply README.md:347]: Agreed as well!

File: stylelint.config.js (hunk 4 above)
Comment (eduardoformiga) [reply stylelint.config.js:36]: I like the #1 better as well 😄

**Review by eduardoformiga — COMMENTED** [review]
LGTM! after applying some suggested changes 👏

File: README.md (hunk 1 above)
Comment (renatamottam) [reply README.md:77]: @filipewl could you try to do it? When I save the file que bar is added automatically :(

File: stylelint.config.js (hunk 4 above)
Comment (renatamottam) [reply stylelint.config.js:36]: The first one shows this error:
image
And when I try to add the extra line requested and save it (see image below), it moves the line back to the beginning.
Screen Shot 2022-01-19 at 09 25 50

Comment (filipewl) [reply stylelint.config.js:36]: Hmm, there seem to be some conflicting Stylelint rules. I was able to commit after the following change to [declaration-colon-space-after](https://stylelint.io/user-guide/rules/list/declaration-colon-space-after/#always-single-line):
CleanShot 2022-01-19 at 15 03 43

File: README.md (hunk 1 above)
Comment (hellofanny) [reply README.md:77]: the same thing is happening here :(

Comment (filipewl) [reply README.md]: There's some tooling conflict that is adding these `\` back... To simplify things I [removed](https://github.com/vtex-sites/base.store/pull/224/commits/214cf00bdba86f3a07159a059cceaf9746a37b9b) both `\` and the _italiczation_ via `_`.

File: README.md [hunk 11]
```diff
@@ -212,108 +215,170 @@ import type { ButtonProps } from '@faststore/ui'
...
 
-function Button (props: Props) {
+function Button(props: Props) {
   return <UIButton {...props} />
 }
 
 export default Button
 ```
 
 For most components, you would stop here. However, buttons can have different variants. For instance, suppose you want to have a button component with primary and muted variants. To add variants to the component, update `Button.tsx`:
```
Comment (renatamottam) [inline README.md:261]:
```suggestion
For most components, you would stop here. However, buttons can have different variants. For instance, suppose you want to have a button component with primary and secondary variants. To add variants to the component, update `Button.tsx`:
```

Comment (hellofanny) [reply README.md]: Nice catch! hihihi

**Review by eduardoformiga — APPROVED** [review]

**Review by hellofanny — APPROVED** [review]

**Review by hellofanny — APPROVED** [review]
